src/main/java/org/elasticsearch/discovery/srv/SrvUnicastHostsProvider.java

```diff
@@ -122,6 +122,13 @@ protected Resolver buildResolver(Settings settings) {
                 logger.warn("Could not create resolver. Using default resolver.", e);
             }
         }
+
+        parent_resolver = parent_resolver == null ? Lookup.getDefaultResolver() : parent_resolver;
```

Comment:
Looking at [the source for Lookup](http://www.dnsjava.org/dnsjava-current/org/xbill/DNS/Lookup.java), it is a shared one.

I can't think of a case where setting the protocol for the default resolver would not be intended, can you? However, I can imagine wanting to set the protocol per server. Maybe that could be an enhancement for the future.